src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForMultipleBuildingBlocks.cs

```diff
@@ -0,0 +1,134 @@
+            IInteractionTasksForChildren<Module, ParameterValuesBuildingBlock> interactionTasksForParameterValues,
+            IInteractionTasksForChildren<Module, MoBiReactionBuildingBlock> interactionTasksForMobiReaction,
+            IInteractionTasksForChildren<Module, MoBiSpatialStructure> interactionTasksForMobiSpatialStructure,
+            IInteractionTasksForChildren<Module, ObserverBuildingBlock> interactionTasksForObserver,
+            IInteractionTaskContext interactionTaskContext)
+         {
+            _interactionTasksForApplicationMoleculeBuilder = interactionTasksForApplicationMoleculeBuilder;
+            _interactionTasksForInitialCondition = interactionTasksForInitialCondition;
+            _interactionTasksForMolecule = interactionTasksForMolecule;
+            _interactionTasksForPassiveTransport = interactionTasksForPassiveTransport;
+            _interactionTasksForParameterValues = interactionTasksForParameterValues;
+            _interactionTasksForMobiReaction = interactionTasksForMobiReaction;
+            _interactionTasksForMobiSpatialStructure = interactionTasksForMobiSpatialStructure;
+            _interactionTasksForObserver = interactionTasksForObserver;
+            _interactionTaskContext = interactionTaskContext;
+         }
+
+         public void RemoveBuildingBlocks(IReadOnlyList<IBuildingBlock> buildingBlocksToRemove)
+         {
+            if (_interactionTaskContext.DialogCreator.MessageBoxYesNo(AppConstants.Dialog.RemoveMultipleBuildingBlocks) != ViewResult.Yes)
+               return;
+
+            var referringSimulationsAndBuildingBlocks = new List<Tuple<IReadOnlyList<IMoBiSimulation>, IBuildingBlock>>();
+
+            foreach (var buildingBlockToRemove in buildingBlocksToRemove)
+            {
+               var referringSimulations = _interactionTaskContext.Context.CurrentProject.SimulationsUsing(buildingBlockToRemove);
+
+               if (referringSimulations.Any())
+                  referringSimulationsAndBuildingBlocks.Add(new Tuple<IReadOnlyList<IMoBiSimulation>, IBuildingBlock>(referringSimulations, buildingBlockToRemove));
+               else
+               {
+                  removeBuildingBlock(buildingBlockToRemove);
+               }
+            }
+
+            if (referringSimulationsAndBuildingBlocks.Any())
+            {
+               var messageBuilder = new StringBuilder();
+
+               messageBuilder.AppendLine(AppConstants.Dialog.BuildingBlocksUsedInSimulation);
+               messageBuilder.AppendLine();
+
+               foreach (var item in referringSimulationsAndBuildingBlocks)
+               {
+                  var buildingBlock = item.Item2;
+                  var simulations = item.Item1;
+
+                  messageBuilder.AppendLine($"\t- {AppConstants.CannotRemoveBuildingBlockFromProject(buildingBlock.Name, simulations.AllNames())}");
+               }
```

Comment:
Can you move all this string manipulation into `AppConstants`. There are a few examples. That removes the complication of building the messages away from this class. As much as possible, pass strings to the AppConstants methods you create rather than the domain objects.